src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -330,16 +353,23 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
             return sniHandle;
         }
 
-        private static byte[] MakeMsSqlServerSPN(string fullyQualifiedDomainName, int port = DefaultSqlServerPort)
+        private static byte[] GetSqlServerSPN(string hostNameOrAddress, string instanceName = null)
         {
-            string serverSpn = SqlServerSpnHeader + "/" + fullyQualifiedDomainName + ":" + port;
+            Debug.Assert(!string.IsNullOrWhiteSpace(hostNameOrAddress));
+            IPHostEntry hostEntry = Dns.GetHostEntry(hostNameOrAddress);
+            string fullyQualifiedDomainName = hostEntry.HostName;
+            if (string.IsNullOrWhiteSpace(instanceName))
+            {
+                instanceName = DefaultSqlServerPort.ToString();
+            }
+            string serverSpn = SqlServerSpnHeader + "/" + fullyQualifiedDomainName + ":" + instanceName;
             return Encoding.UTF8.GetBytes(serverSpn);
         }
 
-        private static string GetFullyQualifiedDomainName(string hostNameOrAddress)
+        private static byte[] GetSqlServerSPN(string hostNameOrAddress, int port)
         {
-            IPHostEntry hostEntry = Dns.GetHostEntry(hostNameOrAddress);
-            return hostEntry.HostName;
+            Debug.Assert(!string.IsNullOrWhiteSpace(hostNameOrAddress) && port >= 0 && port <= 65535);
```

Comment:
Do we need this validation for port? We shouldn't care for the network details about the range of the port. Let TCP connection fail at a later stage with the details that it needs to fail.
The Assert seems further misplaced for SPN generation where a port number will simply be appended.